src/PlaywrightSharp.Tests/Page/PageWaitForRequestTests.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using PlaywrightSharp.Tests.BaseTests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace PlaywrightSharp.Tests.Page
+{
+    ///<playwright-file>page.spec.js</playwright-file>
+    ///<playwright-describe>Page.waitForRequest</playwright-describe>
+    public class PageWaitForRequestTests : PlaywrightSharpPageBaseTest
+    {
+        internal PageWaitForRequestTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        ///<playwright-file>page.spec.js</playwright-file>
+        ///<playwright-describe>Page.waitForRequest</playwright-describe>
+        ///<playwright-it>should work</playwright-it>
+        [Fact]
+        public async Task ShouldWork()
+        {
+            await Page.GoToAsync(TestConstants.EmptyPage);
+            var (request, _) = await TaskUtils.WhenAll(
+                Page.WaitForRequestAsync(TestConstants.ServerUrl + "/digits/2.png"),
+                Page.EvaluateAsync(@"() => {
+                  fetch('/digits/1.png');
+                  fetch('/digits/2.png');
+                  fetch('/digits/3.png');
+                }")
+            );
+            Assert.Equal(TestConstants.ServerUrl + "/digits/2.png", request.Url);
+        }
+
+        ///<playwright-file>page.spec.js</playwright-file>
+        ///<playwright-describe>Page.waitForRequest</playwright-describe>
+        ///<playwright-it>should work with predicate</playwright-it>
+        [Fact]
+        public async Task ShouldWorkWithPredicate()
+        {
+            await Page.GoToAsync(TestConstants.EmptyPage);
+            var (requestEvent, _) = await TaskUtils.WhenAll(
+                Page.WaitForEvent(PageEvent.Request, new WaitForEventOptions<RequestEventArgs>
+                {
+                    Predicate = e => e.Request.Url == TestConstants.ServerUrl + "/digits/2.png"
+                }),
+                Page.EvaluateAsync<string>(@"() => {
+                    fetch('/digits/1.png');
+                    fetch('/digits/2.png');
+                    fetch('/digits/3.png');
+                }")
+            );
+            Assert.Equal(TestConstants.ServerUrl + "/digits/2.png", requestEvent.Request.Url);
+        }
+
+        ///<playwright-file>page.spec.js</playwright-file>
+        ///<playwright-describe>Page.waitForRequest</playwright-describe>
+        ///<playwright-it>should respect timeout</playwright-it>
+        [Fact]
+        public async Task ShouldRespectTimeout()
+        {
+            var exception = await Assert.ThrowsAsync<TimeoutException>(
+                () => Page.WaitForEvent(PageEvent.Request, new WaitForEventOptions<RequestEventArgs>
+                {
+                    Predicate = _ => false,

```

Comment:
Can we avoid this?

Reply:
which part? creating an instance of `WaitForEventOptions` or the predicate that evaluates to `false`?

Reply:
The predicate evaluating to false. We could check for null in the implementation.

Reply:
if it evaluates to `false` then it means that the Task never completes so checking for null won't help up because if we don't pass that property in then it acts as if it was `_ => true` and waits for any event

Reply:
Got it, great job man!